domain/machine/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+	"github.com/juju/utils/v3"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain/life"
+)
+
+// InsertBootstrapMachine inserts the initial machine during bootstrap.
+// TODO - this just creates a minimal row for now.
+func InsertBootstrapMachine(machineId string) func(context.Context, database.TxnRunner) error {
```

Comment:
Can we remove the word Bootstrap? Sort of implied based on the package.

Reply:
Totally understand the concern, but here it literally is called the "bootstrap machine". I could maybe rename the method to InsertMachineZero (which is what it always will be) and lose the machine id param.

Reply:
Went with InsertMachine() for now; can revisit later when it's all fully implemented.